lib/matplotlib/axes/_base.py

```diff
@@ -2185,6 +2186,10 @@ def grid(self, b=None, which='major', axis='both', **kwargs):
             b = True
         b = _string_to_bool(b)
 
+        # We need to disable both grids before replotting the wanted ones
+        self.xaxis.grid(False)
```

Comment:
Alright, I did not understood it that way. It makes sense indeed, although it does not seems the most intuitive in my opinion.